PCbuild/find_python.bat

```diff
@@ -34,10 +34,10 @@
 @if exist "%_Py_EXTERNALS_DIR%\pythonx86\tools\python.exe" (set PYTHON="%_Py_EXTERNALS_DIR%\pythonx86\tools\python.exe") & (set _Py_Python_Source=found in externals directory) & goto :found
```

Comment:
```suggestion
@if exist "%_Py_EXTERNALS_DIR%\pythonx86\tools\python.exe" ("%_Py_EXTERNALS_DIR%\pythonx86\tools\python.exe" -Ec "import sys; assert sys.version_info[:2] >= (3, 10)" >nul 2>nul) && (set PYTHON="%_Py_EXTERNALS_DIR%\pythonx86\tools\python.exe") && (set _Py_Python_Source=found in externals directory) && goto :found || rmdir /Q /S "%_Py_EXTERNALS_DIR%\pythonx86"
```

I've tested this, so should be good to just merge. If you want to try it yourself, easiest way is to bump the assertion in this statement so that it'll even fail on 3.10, and comment out line 40. That should force it to delete any downloaded Python runtime and download it again.

Reply:
I merged it, but with the wrong commit message -- I'll amend it now.